protostar/cli/argument.py

```diff
@@ -0,0 +1,20 @@
+from dataclasses import dataclass, replace
+from typing import Any, Generic, Optional, TypeVar
+
+ArgTypeNameT = TypeVar("ArgTypeNameT")
+
+# pylint: disable=too-many-instance-attributes
+@dataclass(frozen=True)
+class Argument(Generic[ArgTypeNameT]):
+    name: str
+    description: str
+    type: ArgTypeNameT
+    is_positional: bool = False
+    is_required: bool = False
+    is_array: bool = False
+    default: Any = None
+    example: Optional[str] = None
+    short_name: Optional[str] = None
+
+    def copy_with(self, **changes) -> "Argument":
```

Comment:
1. hmm why not simply use `dataclasses.replace` everywhere by convention?
2. if not, could this return `-> Self` ?

Reply:
It's old code.
1) Flutter's `copy_with` seemed more natural to me back then.
2) I didn't know about `Self` type back then.

I try to keep this PR focused.

Reply:
ok, gotcha